website/blog/2022-07-27-getting-started-with-the-dbt-semantic-layer.md

```diff
@@ -0,0 +1,165 @@
+
+***What will this look like for my data consumers and business stakeholders?*** 
+
+Ultimately, this looks like people being able to interact with trusted datasets in the tools that they are comfortable with (and eventually new tools designed specifically around metrics). 
+
+An example that we’ve found helpful is [ARR](https://www.zuora.com/billing-topics/annual-recurring-revenue/#:~:text=Annual%20Recurring%20Revenue%2C%20or%20ARR,for%20a%20single%20calendar%20year). A business-critical metric to SaaS companies, ARR can be a tricky calculation to keep consistent across all of the tools used in the business. With the semantic layer, this definition would live in dbt and the logic to create the dataset for that metric would be consistent across all different consuming experiences. Best of all, definition changes would get reflected in downstream tools, so you no longer need to manually search and update every downstream dependency. Callum of 3 years ago is jumping with joy.
+
+***That’s good and all, but what does this look like for practitioners to use?*** 
+
+We’re building it with the following components[^1]:
+
+**Available Today**
+
+- `**[metric` node in `dbt-core` :](https://docs.getdbt.com/docs/building-a-dbt-project/metrics)** Similar to `models` or `sources` , this is a specific node type in `dbt-core`. It is the definition of a time-series aggregation over a table that supports 0+ dimensions. The resulting metadata is stored in the `manifest` just like `models` and referenced in the DAG.
+- `**[dbt_metrics` package:](https://github.com/dbt-labs/dbt_metrics)** this package provides macros that take in metric information regarding the `metrics` defined by users and then produces the necessary SQL to return the dataset containing that metric along with any other parameters defined (secondary calculations, specific dimensions, time grain, etc).
+- **[dbt Cloud Metadata API](https://docs.getdbt.com/docs/dbt-cloud/dbt-cloud-api/metadata/metadata-overview):** a GraphQL API which supports arbitrary queries over the metadata produced by dbt Cloud jobs. Contains metadata related to the accuracy, recency, configuration, and structure of the views and tables in the warehouse, as well as much more.
+
+**Forthcoming**
+
+- **dbt Server:** this component wraps dbt Core in a persistent server that is responsible for handling RESTful API requests for dbt operations. It’s a thin interface that is primarily responsible for performance and reliability in production environments.
+- **dbt Cloud proxy server:** this component enables dbt Cloud to intercept requests to a database and compile dbt-SQL into raw SQL that the database understands. It then returns the dataset produced by the raw SQL to the platform that sent it.
+
+![Untitled](/img/blog/2022-07-27-getting-started-with-the-dbt-semantic-layer/semantic-layer-description.png)
+
+### Understanding how and when to use metrics?
+
```

Comment:
Wonder if some people reading this might not be familiar with the jaffle shop. Could we say something along the lines of "Let's walk through an example of how you can use the components above to get started today using our old friend - the Jaffle Shop [link]". We'll take a look at how you can start defining and testing metrics today as well as how you'll interact with them once the forthcoming components are released.

Reply:
Updated